sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkTracerProviderBuilder.java

```diff
@@ -68,6 +73,40 @@ public SdkTracerProviderBuilder setIdGenerator(IdGenerator idGenerator) {
     return this;
   }
 
+  /**
+   * Assign an {@link EntityProvider} which can generate a Resource to be attached to all Spans
+   * created by Tracers.
+   *
+   * @param entityProvider An EntityProvider implementation.
+   * @return this
+   */
+  public SdkTracerProviderBuilder setEntityProvider(EntityProvider entityProvider) {
+    requireNonNull(entityProvider, "entityProvider");
+    this.entityProvider = entityProvider;
+    return this;
+  }
+
+  /**
+   * Merge an {@link Entity} into the EntityProvider.
+   *
+   * @since ??.??.??
+   */
+  public SdkTracerProviderBuilder addEntity(String id, String name, Attributes attributes) {
```

Comment:
Is the plan to expose this on TraceProvider?

Why not force things through EntityProvider?

Reply:
Oh, I dunno....this is almost certainly misplaced, good catch. I think I should remove it.

I think maybe I was just hammering through all the compilation problems and thought this needed parity with the `addResource()` but that's not quite right.

Reply:
I'm still not sure how the user is supposed to get ahold of the `EntityProvider` though.

Reply:
Yeah - I'm thinking we may need it to be an API level thing ... But hoping prototypes like this can teach us that.